src/ann_solo/reader.py

```diff
@@ -231,6 +299,324 @@ def get_version(self) -> str:
+        identifier : List[int]
+            Incremented identifier of the spectrum in the library.
+
+        Returns
+        -------
+        Iterator[MsmsSpectrum]
+            An iterator of spectra in the given library file.
+        """
+        for spectrum in list(Parallel(n_jobs=self._num_threads)(
+                delayed(self._parse_sptxt_spectrum)(raw_spectrum, id) for
+                raw_spectrum, id in zip(raw_spectra, ids))):
+            yield spectrum
+
+    def parse_sptxt(self) -> Iterator[MsmsSpectrum]:
+        """
+        Open the sptxt spectra library file and parses it
+        to read all spectra.
+
+        Returns
+        -------
+        Iterator[MsmsSpectrum]
+            An iterator of spectra in the given library file.
+
+        """
+        with open(self._filename, 'rb') as file:
+            mmapped_file = mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ)
+            raw_spectra_list = []
+            for i, raw_spec in enumerate(re.finditer(
+                    b'(?<![a-zA-Z])Name:\s?(?:(?!((?<![a-zA-Z])Name:\s?)).|\n)*',
+                        mmapped_file.read(),
+                        re.IGNORECASE),
+                    1):
+                raw_spectra_list.append(
+                    '\n'.join(raw_spec.group(0).decode('utf-8').splitlines()))
+                if i % self._num_threads:
+                    continue
+                else:
+                    for spectrum in self._parse_batch_sptxt_spectra(
+                            raw_spectra_list,
+                            list(range(i + 1 - self._num_threads, i + 1))):
+                        yield spectrum
+                    raw_spectra_list = []
+            if not raw_spectra_list:
+                for spectrum in \
+                    self._parse_batch_sptxt_spectra(raw_spectra_list,
+                                            list(range(
+                                                i + 1 - len(raw_spectra_list),
+                                                i + 1))
+                                            ):
+                    yield spectrum
+
+
+
+
+class SpectraLibraryStore:
+    """
+        Class to efficiently store and retrieve spectra from a library file.
+    """
+    def __init__(self, file_path: str) -> None:
+        """
+        Initialize the spectral library store.
+
+        Parameters
+        ----------
+        filepath : str
+            The file path of the spectral library store.
+
+        """
+        self.file_path = file_path
+        self.hdf5_store = None
+
+    def open_store(self,mode: str) -> None:
+        """
+        Open the hdf5 spectral library store for read/write purposes.
+
+        Parameters
+        ----------
+        mode : str
+            The mode (r/w) by which to open the  spectral library store.
+        """
+        self.hdf5_store = h5py.File(self.file_path,
+                                    'r' if mode == 'r' else 'w')
```

Comment:
**todo:** Just provide the mode directly. The docstring specifies that it should be either "r" or "w", so the if statement doesn't have an effect.